pages/slice/language-reference/slice-grammar.md

```diff
@@ -2,14 +2,587 @@
+Base classes are specified by a single colon, followed by a class's (possibly scoped) identifier.
+Class bodies consist of a list of [fields][field] wrapped in a pair of braces. These fields may be optionally separated by commas.
+Additionally, [local attributes][attribute] and a [doc-comment] may be applied to the class in its prelude.
+
+```ebnf {% showTitle=false %}
+Class
+    : Prelude "class" Identifier CompactId? (":" TypeRef)? "{" UndelimitedList<Field> "}"
+    ;
+
+CompactId
+    : "(" SignedInteger ")"
+    ;
+```
+
+For additional information on classes, see the [class][class-guide] page.
+
+### Exceptions
+
+Exceptions can only be defined or referenced in [`Slice1`][compilation-mode-guide] mode.
+Exceptions cannot be used as types. They can only validly appear in the exception specification of [operations][operation].
+
+Exception definitions consist of the `exception` keyword, followed by an [identifier], optionally followed by a base exception, and then the exception's body.
+Base exceptions are specified by a single colon, followed by an exception's (possibly scoped) identifier.
+Exception bodies consist of a list of [fields][field] wrapped in a pair of braces. These fields may be optionally separated by commas.
+Additionally, [local attributes][attribute] and a [doc-comment] may be applied to the exception in its prelude.
+
+```ebnf {% showTitle=false %}
+Exception
+    : Prelude "exception" Identifier (":" TypeRef)? "{" UndelimitedList<Field> "}"
+    ;
+```
+
+For additional information on exceptions, see the [exception][exception-guide] page.
+
+### Fields
+
+Fields can only be declared within [structs][struct], [classes][class], and [exceptions][exception].
+Field declarations consist of an [identifier], followed by a colon, and then a [type].
+Optionally, a [tag] may be applied to the field, directly before its identifier, making this a [tagged field][tag].
+Additionally, [local attributes][attribute] and a [doc-comment][doc-comment] may be applied to the field in its prelude.
+
+```ebnf {% showTitle=false %}
+Field
+    : Prelude Tag? Identifier ":" TypeRef
+    ;
+```
+
+For additional information on fields, see the [field][field-guide] page.
+
+### Interfaces
+
+Interface definitions consist of the `interface` keyword, followed by an [identifier], optionally followed by a list of base interfaces, and then the interface's body.
+Base interfaces are specified by a single colon, followed by the (possibly scoped) identifiers of one or more interfaces, separated by commas.
+Interface bodies consist of a set of [operations][operation] wrapped in a pair of braces.
+Additionally, [local attributes][attribute] and a [doc-comment] may be applied to the class in its prelude.
+
+```ebnf {% showTitle=false %}
+Interface
+    : Prelude "interface" Identifier (":" NonEmptyCommaList<TypeRef>)? "{" Operation* "}"
+    ;
+```
+
+If you are using an RPC framework, every interface definition also defines a _proxy type_.
```

Comment:
I'm fine removing this whole paragraph on proxies. I also find it out of place, and frankly confusing.

I only added it because we have an entire page for proxies in the language guide, and we draw a big line between interfaces and proxies. Should we do anything about that page?